src/components/structures/MatrixChat.tsx

```diff
@@ -1949,21 +1957,32 @@ export default class MatrixChat extends React.PureComponent<IProps, IState> {
         });
     }
 
-    private setPageSubtitle(subtitle = ""): void {
+    private setPageSubtitle(): void {
+        const params: {
+            $brand: string;
+            $status: string;
+            $room_name: string|undefined;
+        } = {
+            $brand: SdkConfig.get().brand,
+            $status: this.subTitleStatus,
+            $room_name: undefined,
+        };
+
         if (this.state.currentRoomId) {
             const client = MatrixClientPeg.get();
             const room = client?.getRoom(this.state.currentRoomId);
             if (room) {
-                subtitle = `${this.subTitleStatus} | ${room.name} ${subtitle}`;
+                params.$room_name = room.name;
             }
-        } else {
-            subtitle = `${this.subTitleStatus} ${subtitle}`;
         }
+        
+        const titleTemplate = params.$room_name ? this.titleTemplateInRoom : this.titleTemplate;
 
-        const title = `${SdkConfig.get().brand} ${subtitle}`;
+        const title = Object.entries(params).reduce(
```

Comment:
:warning: I wrote my own templating function here.

I feel like an app this size must be using templating elsewhere but couldn't find any examples or libraries I was particularly keen on, but pointers in this direction would be appreciated.